File: python/threads/threads.py
import threading
import time


def perform_simple_operation():
    while True:
        _ = 42 * 42
        time.sleep(1)


def main():
    num_threads = 1000000
    threads = []

    for i in range(num_threads):
        thread = threading.Thread(target=perform_simple_operation)
        thread.start()
        threads.append(thread)

    for thread in threads:
        thread.join()


if __name__ == "__main__":
    main()

File: c#/threads/Threads.cs
using System;
using System.Threading;
using System.Collections.Generic;

class Threads
{
    static void PerformSimpleOperation()
    {
        while (true)
        {
            var result = 42 * 42;
            Thread.Sleep(1000); // 1000 milliseconds = 1 second
        }
    }

    static void Main(string[] args)
    {
        int numThreads = 1;
        List<Thread> threads = new List<Thread>();

        for (int i = 0; i < numThreads; i++)
        {
            Thread thread = new Thread(PerformSimpleOperation);
            thread.Start();
            threads.Add(thread);
        }

        foreach (Thread thread in threads)
        {
            thread.Join();
        }
    }
}
